core/management/commands/update_country_regions.py

```diff
@@ -0,0 +1,532 @@
+
+import requests
+from django_countries.data import COUNTRIES
+from unidecode import unidecode
+
+from hosting.countries import COUNTRIES_DATA
+from hosting.models import LOCATION_REGION, CountryRegion, Whereabouts
+from maps import SRID
+
+GEONAMES_SOURCE_URL = (
+    'http://api.geonames.org/searchJSON?country={country}&featureCode={code}'
+    '&style=FULL&lang=eo&maxRows=1000&username=pasportaservo'
+)
+GEONAMES_DETAILS_SOURCE_URL = (
+    'http://api.geonames.org/getJSON?geonameId={}&username=pasportaservo'
+)
+COMMERCEGUYS_SOURCE_URL = (
+    'https://raw.githubusercontent.com/commerceguys/addressing'
+    '/master/resources/subdivision/{}.json'
+)
+
+
+class Command(BaseCommand):
+    help = """
+        Updates the administrative units (e.g., states or provinces) for a country.
+        """
+
+    def add_arguments(self, parser):
+        """
+        The command can be given a list of ISO country codes to update,
+        as an argument. A missing argument or the word "ALL" mean that
+        all countries in the local database will be refreshed.
+        """
+        parser.add_argument(
+            'update_countries',
+            nargs='*', choices=tuple(sorted(COUNTRIES.keys())) + ('ALL',), default='ALL',
+            metavar='country code',
+            help="one or more country codes to update, omit to update all countries.")
+        parser.add_argument(
+            '-s', '--start-from',
+            default='AA', choices='ABCDEFGHIJKLMNOPQRSTUVWXYZ',
+            help="when all countries are being updated, begin with the ISO code "
+                 "starting with this letter.")
+        parser.add_argument(
+            '-e', '--end-before',
+            default='ZZ', choices='ABCDEFGHIJKLMNOPQRSTUVWXYZ',
+            help="when all countries are being updated, stop before the ISO code "
+                 "starting with this letter.")
+
+    TOPONYM_REPLACEMENTS = {
+        "EG": "Governorate",
+        "IR": "Province",
+        "MP": "Municipality",
+        "NR": "District",
+        "TW": "County",
+        "VN": ["Province", "City"],
+    }
+
+    def handle(self, *args, **options):
+        self.verbosity = options['verbosity']
+        self.update_countries = update_countries = options['update_countries']
+        if not update_countries or update_countries == 'ALL' or 'ALL' in update_countries:
+            update_countries = tuple(
+                c for c in sorted(COUNTRIES.keys())
+                if c >= options['start_from'] and c < options['end_before']
+            )
+            self.update_countries = list(sorted(COUNTRIES.keys()))
+            self.full_update = True
+        else:
+            self.full_update = False
+
+        country_result, country_removals = {}, {}
+        for country_code in update_countries:
+            # First try to retrieve the high-quality data from CommerceGuys.
+            result, json_data = self.load_commerceguys_data(country_code)
+            if not result:
```

Comment:
Kial ne `if json_data is None` kaj ne nur redoni `None` se `load_commerceguys_data` fuŝis?

Reply:
ĉar la biblioteko _addressing_ ne havas la detalojn de ĉiuj landoj kiuj postulas regionon – por iuj landoj `administrativeArea` estas inter `required_fields` (vidu `countries.py`) sed ili ne aperas en la datumoj de la biblioteko. do povas esti tri rezultoj de `load_commerce_guys`: aŭ 200 => json aŭ 404 => `None` aŭ io alia => ŝargado fuŝis.